### ISSUE TITLE
[bug] Cannot get features because assertion cannot be executed

### ISSUE THREAD

**Issue by AhmetNSimsek** [FZJ-INM1-BDA/siibra-python#234]
TypeError (Subscripted generics cannot be used with class and instance checks) at
AnatomicalAnchor.__init__
line 112: `assert isinstance(region, Union[str, None])`
(version: refactor_v0.4a16)

From what I can tell, this seems to be an issue after python 3.7 (inclusive).

```python
import siibra
from os import environ
environ['SIIBRA_ENABLE_DEVICE_FLOW'] = "1"
siibra.fetch_ebrains_token()
atlas = siibra.atlases.MULTILEVEL_HUMAN_ATLAS
p = atlas.parcellations["Julich"]
region = p.get_region("v2")
features = siibra.get_features(concept=atlas, modality=siibra.modalities.EbrainsAnchoredDataset, region=region)
```

**Comment by dickscheid** [comment]
There are two issues here:

1. The `get_features` method does not expect an atlas as concept; typically this is a region or parcellation -> needs to be catched upfront, we will fix that.
2. The addition region= parameter is not expected, but accepted since `get_features` accepts **kwargs - however it seems not be the cause of the error.
3. Indeed is instance does not accept generic type specs in Python 3.7. We will change to a simple tuple.

**Comment by dickscheid** [comment]
Should be fixed by 01e9b4a - could you verify?

**Comment by AhmetNSimsek** [comment]
It is now fixed indeed.
